sdk/python/kfp/components/structures.py

```diff
@@ -931,12 +942,11 @@ def to_pipeline_spec(self) -> pipeline_spec_pb2.PipelineSpec:
         # can just assign the component_spec_proto directly to .root
         component_spec_proto = builder._build_component_spec_from_component_spec_structure(
             self)
-        has_inputs = bool(
-            len(component_spec_proto.input_definitions.artifacts) +
-            len(component_spec_proto.input_definitions.parameters))
-        if has_inputs:
-            pipeline_spec.root.input_definitions.CopyFrom(
-                component_spec_proto.input_definitions)
+        pipeline_spec.root.CopyFrom(component_spec_proto)
+
+        if pipeline_outputs:
```

Comment:
Is `if pipeline_outputs` necessary? I think the `_builder_dag_outputs` generalizes to the case of an empty `dag_outputs` dict.